src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -61,11 +65,13 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             Document document, SyntaxEditor editor,
             SyntaxNode node, CancellationToken cancellationToken)
         {
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var declarationContext = node.Parent;
 
-            TypeSyntax typeSyntax = null;
-            ParenthesizedVariableDesignationSyntax parensDesignation = null;
+            TypeSyntax? typeSyntax = null;
+            ParenthesizedVariableDesignationSyntax? parensDesignation = null;
+            SyntaxNode? declarationSyntax = null;
```

Comment:
i officially state hat i hate this method.  it's so hard ot understand all teh data flow because it attempts to branch out to multiple different conditions, but hten collapse back on common code that then goes and checks all those different conditions.

Reply:
I'll split this up